aten/src/ATen/native/cuda/TensorTopK.cu

```diff
@@ -0,0 +1,338 @@
+    at::cuda::detail::TensorInfo<scalar_t, INDEX_T> topKInfo =            \
+      at::cuda::detail::getTensorInfo<scalar_t, INDEX_T>(values);         \
+    at::cuda::detail::TensorInfo<int64_t, INDEX_T> indicesInfo =          \
+      at::cuda::detail::getTensorInfo<int64_t, INDEX_T>(indices);         \
+    /* tensorInfoLegacyIfScalar*/                                         \
+    if (!input.dim()) {                                                   \
+      inputInfo.dims = 1;                                                 \
+      inputInfo.sizes[0] = 1;                                             \
+      inputInfo.strides[0] = 1;                                           \
+      topKInfo.dims = 1;                                                  \
+      topKInfo.sizes[0] = 1;                                              \
+      topKInfo.strides[0] = 1;                                            \
+      indicesInfo.dims = 1;                                               \
+      indicesInfo.sizes[0] = 1;                                           \
+      indicesInfo.strides[0] = 1;                                         \
+    }                                                                     \
+    /* We use these structures solely to find the offset to */            \
+    /* each slice we are operating on */                                  \
+    inputInfo.sizes[dim] = 1;                                             \
+    topKInfo.sizes[dim] = 1;                                              \
+    indicesInfo.sizes[dim] = 1;                                           \
+    /* Collapse all other dims */                                         \
+    int collapseInputDim = inputInfo.collapseDims(dim);                   \
+    int collapseTopKDim = topKInfo.collapseDims(dim);                     \
+    int collapseIndicesDim = indicesInfo.collapseDims(dim);               \
+    int64_t inputSlices = 1;                                              \
+    for (int i = 0; i < inputInfo.dims; ++i) {                            \
+      inputSlices *= inputInfo.sizes[i];                                  \
+    }                                                                     \
+    int64_t topKSlices = 1;                                               \
+    for (int i = 0; i < topKInfo.dims; ++i) {                             \
+      topKSlices *= topKInfo.sizes[i];                                    \
+    }                                                                     \
+                                                                          \
+    dim3 grid;                                                            \
+    if (!THC_getGridFromTiles(inputSlices, grid)) {                       \
+      THError("Slice to sort is too large");                              \
+    }                                                                     \
```

Comment:
Please use `TORCH_CHECK`

Reply:
(since this is the same pattern as before I'll replace it with TORCH_INTERNAL_ASSERT)